admiral/envs/components/examples/bird_flight.py

```diff
@@ -0,0 +1,111 @@
+
+from matplotlib import pyplot as plt
+import numpy as np
+
+from admiral.envs.components.agent import PositionAgent, SpeedAngleAgent
+from admiral.envs.components.state import ContinuousPositionState, SpeedAngleState
+from admiral.envs.components.actor import SpeedAngleMovementActor
+from admiral.envs.components.observer import SpeedObserver, AngleObserver
+from admiral.envs.components.done import TooCloseDone
+from admiral.envs import AgentBasedSimulation
+from admiral.tools.matplotlib_utils import mscatter
+
+class BirdAgent(PositionAgent, SpeedAngleAgent): pass
+
+class Flight(AgentBasedSimulation):
+    def __init__(self, **kwargs):
+        self.agents = kwargs['agents']
+
+        # State
+        self.position = ContinuousPositionState(**kwargs)
+        self.speed_angle = SpeedAngleState(**kwargs)
+
+        # Actor
+        self.move_actor = SpeedAngleMovementActor(position=self.position, speed_angle=self.speed_angle, **kwargs)
+
+        # Observer
+        self.speed_observer = SpeedObserver(**kwargs)
+        self.angle_observer = AngleObserver(**kwargs)
+
+        # Done
+        self.done = TooCloseDone(position=self.position, **kwargs)
+
+        from gym.spaces import MultiBinary
+        for agent in self.agents.values():
+            agent.observation_space = {'dummy': MultiBinary(1)}
+        self.finalize()
+
+    def reset(self, **kwargs):
+        self.position.reset(**kwargs)
+        self.speed_angle.reset(**kwargs)
+    
+    def step(self, action_dict, **kwargs):
+        for agent, action in action_dict.items():
+            self.move_actor.move(self.agents[agent], action['accelerate'][0], action['banking_angle_change'][0], **kwargs)
```

Comment:
Should not require environments to access the 0th element of the actions. It would be better to just have this incoming as an array and process it as such in the move_actor's move function.